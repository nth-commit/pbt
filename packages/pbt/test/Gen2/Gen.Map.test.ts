import fc from 'fast-check';
import * as dev from './srcShim';
import * as domainGen from './Helpers/domainGen';

test('sample(gen.map(f)).values = sample(gen).values.map(f)', () => {
  fc.assert(
    fc.property(domainGen.sampleConfig(), domainGen.gen(), fc.func(fc.anything()), (config, gen, f) => {
      const genMapped = gen.map((x) => f(x));

      const mappedSample = dev.sampleTrees(genMapped, config);
      const unmappedSample = dev.sampleTrees(gen, config);

      const actualTrees = mappedSample.values.map(dev.GenTree.traverseGreedy);
      const expectedTrees = unmappedSample.values
        .map((tree) => dev.GenTree.map(tree, f))
        .map(dev.GenTree.traverseGreedy);
      expect(actualTrees).toEqual(expectedTrees);
    }),
  );
});
