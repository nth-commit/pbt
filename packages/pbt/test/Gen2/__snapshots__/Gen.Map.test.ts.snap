// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: 0 1`] = `"A (c = 0)"`;

exports[`snapshot: 1 1`] = `
"B (c = 10)
..A (c = 0)"
`;

exports[`snapshot: 2 1`] = `
"B (c = 10)
..A (c = 0)"
`;

exports[`snapshot: 3 1`] = `
"D (c = 30)
..A (c = 0)
..B (c = 10)
....A (c = 0)
..C (c = 20)
....A (c = 0)
....B (c = 10)
......A (c = 0)"
`;

exports[`snapshot: 4 1`] = `
"B (c = 10)
..A (c = 0)"
`;

exports[`snapshot: 5 1`] = `
"B (c = 10)
..A (c = 0)"
`;

exports[`snapshot: 6 1`] = `"A (c = 0)"`;

exports[`snapshot: 7 1`] = `
"H (c = 70)
..A (c = 0)
..D (c = 30)
....A (c = 0)
....B (c = 10)
......A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
..F (c = 50)
....A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
....D (c = 30)
......A (c = 0)
......B (c = 10)
........A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
....E (c = 40)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
..G (c = 60)
....A (c = 0)
....D (c = 30)
......A (c = 0)
......B (c = 10)
........A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
....E (c = 40)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
....F (c = 50)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
......E (c = 40)
........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
........D (c = 30)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
..........C (c = 20)
............A (c = 0)
............B (c = 10)
..............A (c = 0)"
`;

exports[`snapshot: 8 1`] = `
"H (c = 70)
..A (c = 0)
..D (c = 30)
....A (c = 0)
....B (c = 10)
......A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
..F (c = 50)
....A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
....D (c = 30)
......A (c = 0)
......B (c = 10)
........A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
....E (c = 40)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
..G (c = 60)
....A (c = 0)
....D (c = 30)
......A (c = 0)
......B (c = 10)
........A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
....E (c = 40)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
....F (c = 50)
......A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
......D (c = 30)
........A (c = 0)
........B (c = 10)
..........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
......E (c = 40)
........A (c = 0)
........C (c = 20)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
........D (c = 30)
..........A (c = 0)
..........B (c = 10)
............A (c = 0)
..........C (c = 20)
............A (c = 0)
............B (c = 10)
..............A (c = 0)"
`;

exports[`snapshot: 9 1`] = `
"B (c = 10)
..A (c = 0)"
`;

exports[`snapshot: 10 1`] = `
"F (c = 50)
..A (c = 0)
..C (c = 20)
....A (c = 0)
....B (c = 10)
......A (c = 0)
..D (c = 30)
....A (c = 0)
....B (c = 10)
......A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
..E (c = 40)
....A (c = 0)
....C (c = 20)
......A (c = 0)
......B (c = 10)
........A (c = 0)
....D (c = 30)
......A (c = 0)
......B (c = 10)
........A (c = 0)
......C (c = 20)
........A (c = 0)
........B (c = 10)
..........A (c = 0)"
`;
